x-pack/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/api/perform_rule_upgrade/perform_rule_upgrade_route.ts

```diff
@@ -0,0 +1,176 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { transformError } from '@kbn/securitysolution-es-utils';
+import { PERFORM_RULE_UPGRADE_URL } from '../../../../../../common/detection_engine/prebuilt_rules';
+import {
+  PerformRuleUpgradeRequestBody,
+  PickVersionValues,
+} from '../../../../../../common/detection_engine/prebuilt_rules/api/perform_rule_upgrade/perform_rule_upgrade_request_schema';
+import type {
+  PerformRuleUpgradeResponseBody,
+  SkippedRuleUpgrade,
+} from '../../../../../../common/detection_engine/prebuilt_rules/api/perform_rule_upgrade/perform_rule_upgrade_response_schema';
+import { SkipRuleUpgradeReason } from '../../../../../../common/detection_engine/prebuilt_rules/api/perform_rule_upgrade/perform_rule_upgrade_response_schema';
+import { assertUnreachable } from '../../../../../../common/utility_types';
+import type { SecuritySolutionPluginRouter } from '../../../../../types';
+import { buildRouteValidation } from '../../../../../utils/build_validation/route_validation';
+import type { PromisePoolError } from '../../../../../utils/promise_pool';
+import { buildSiemResponse } from '../../../routes/utils';
+import { internalRuleToAPIResponse } from '../../../rule_management/normalization/rule_converters';
+import { aggregatePrebuiltRuleErrors } from '../../logic/aggregate_prebuilt_rule_errors';
+import { createPrebuiltRuleAssetsClient } from '../../logic/rule_assets/prebuilt_rule_assets_client';
+import { createPrebuiltRuleObjectsClient } from '../../logic/rule_objects/prebuilt_rule_objects_client';
+import { upgradePrebuiltRules } from '../../logic/rule_objects/upgrade_prebuilt_rules';
+import { fetchRuleVersionsTriad } from '../../logic/rule_versions/fetch_rule_versions_triad';
+import type { PrebuiltRuleAsset } from '../../model/rule_assets/prebuilt_rule_asset';
+import { getVersionBuckets } from '../../model/rule_versions/get_version_buckets';
+
+export const performRuleUpgradeRoute = (router: SecuritySolutionPluginRouter) => {
+  router.post(
+    {
+      path: PERFORM_RULE_UPGRADE_URL,
+      validate: {
+        body: buildRouteValidation(PerformRuleUpgradeRequestBody),
+      },
+      options: {
+        tags: ['access:securitySolution'],
+      },
+    },
+    async (context, request, response) => {
+      const siemResponse = buildSiemResponse(response);
+
+      try {
+        const ctx = await context.resolve(['core', 'alerting']);
+        const soClient = ctx.core.savedObjects.client;
+        const rulesClient = ctx.alerting.getRulesClient();
+        const ruleAssetsClient = createPrebuiltRuleAssetsClient(soClient);
+        const ruleObjectsClient = createPrebuiltRuleObjectsClient(rulesClient);
+
+        const { mode, pick_version: globalPickVersion = PickVersionValues.TARGET } = request.body;
```

Comment:
So, if I understand correctly, for the current Milestone we won't need to pass the `pick_version` param, since we will always want to update to the target version of the rule.
Is that correct?

Reply:
Yes, that's right. The version to pick will always be `TARGET`.